vmdb/app/models/miq_provision/naming.rb

```diff
@@ -51,28 +51,12 @@ def get_vm_full_name(unresolved_vm_name, prov_obj, determine_index)
       # if we are just building a sample of what the vm_name will look like use '#' inplace of actual number.
       return "#{name[:prefix]}#{'#' * name[:index_length]}#{name[:suffix]}" if determine_index == false
 
-      # Determine starting index based on already assigned names in miq_provision table.
-      start_idx = 0
-      reg_val = Regexp.new("(#{name[:prefix]})(\\d{#{name[:index_length]}})(#{name[:suffix]})")
-
-      MiqProvision.find(:all).each do |p|
-        if p.options[:vm_target_name].to_s.strip =~ reg_val
-          name_prefix, name_idx, name_suffix = $1.to_s, $2.to_i, $3
-          # If the name prefix and suffix match record the highest index number
-          if name[:prefix].downcase == name_prefix.downcase && name[:suffix].to_s.downcase == name_suffix.to_s.downcase
-            start_idx = name_idx if name_idx > start_idx
-          end
-        end
-      end
-      start_idx+=1
-
-      start_idx.upto(9999) do |x|
-        idx_str  = format("%0#{name[:index_length]}d", x)
+      loop do
+        idx_str  = format("%0#{name[:index_length]}d", MiqRegion.my_region.next_naming_index(unresolved_vm_name))
```

Comment:
Funny, I actually prefer the % operator for formatting even though rubocop disagrees...I think we should switch.  This style just looks so non-Ruby-ish...kinda Python-ish actually.